### PR TITLE
chore: Re implement channels in list objects

### PR DESCRIPTION
**Pull request by harshabangi**
Re-implement channels in ListObjects

## Description

1. Create a struct with members objectID and Error and used this struct to pass both object and error in the channels and removed all the usages of separate channels used for object and error.

## References
Addresses issue - https://github.com/openfga/openfga/issues/764

## Review Checklist
- [x] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: pkg/server/commands/list_objects.go [hunk 1]
```diff
@@ -36,6 +36,11 @@ type ListObjectsQuery struct {
 	ResolveNodeLimit      uint32
 }
 
+type ObjectOrError struct {
```
Comment (jon-whit) [inline pkg/server/commands/list_objects.go:39]: Can we please rename to `ListObjectsResult`?

Comment (harshabangi) [reply pkg/server/commands/list_objects.go]: Resolved in https://github.com/openfga/openfga/pull/779/commits/179f7dd2f538126a85d7cdfc59ee06bba53f0590

**Review by miparnisari — COMMENTED** [review]
Tested with https://github.com/uber-go/goleak, no leaks seen!

File: pkg/server/commands/list_objects.go [hunk 2]
```diff
@@ -275,20 +270,20 @@ func (q *ListObjectsQuery) ExecuteStreamed(
 				return nil
 			}
 
+			if object.Err != nil {
```
[the remark this answers is not in the document]
Comment (harshabangi) [reply pkg/server/commands/list_objects.go]: Resolved in https://github.com/openfga/openfga/pull/779/commits/2024c2271d2dcce7346f68f3bba125fa7de78764

**Review by miparnisari — APPROVED** [review]

**Review by elbuo8 — APPROVED** [review]

**Pull request merged**